modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java

```diff
@@ -248,6 +248,16 @@ public CommerceOrderItem addCommerceOrderItem(
 			commerceOrderItemPersistence.update(childCommerceOrderItem);
 		}
 
+		if (commerceOrder.isManuallyAdjusted() && commerceOrder.isOpen()) {
+			commerceOrder.setManuallyAdjusted(false);
+		}
+
+		if (!commerceOrder.isShippable() && commerceOrderItem.isShippable()) {
+			commerceOrder.setShippable(true);
+		}
+
+		commerceOrderLocalService.updateCommerceOrder(commerceOrder);
+
```

Comment:
to avoid concurrency errors during update
```
		CommerceOrder newCommerceOrder = commerceOrder;

		if (commerceOrder.isManuallyAdjusted() && commerceOrder.isOpen()) {
			newCommerceOrder.setManuallyAdjusted(false);
		}

		if (!commerceOrder.isShippable() && commerceOrderItem.isShippable()) {
			newCommerceOrder.setShippable(true);
		}

		if (!Objects.equals(newCommerceOrder, commerceOrder)) {
			commerceOrderLocalService.updateCommerceOrder(newCommerceOrder);
		}
```